### PR TITLE
refactor(react): merge hoc's code into original component file

### PR DESCRIPTION
**Pull request by manudeli**
related #62 

# merge HOC's code into original component file

1. Merge HOCs into original Components to manage files easily
2. Replace all unnecessary complex `ComponentPropsWithoutChildren` type to `PropsWithoutChildren`
3. Generic `T` prefix convention to make it easy to understand itself as Generic
4. Remove many files and unnecessary directories like `index.ts` and `hooks/*` or `utils/*`
5. Add test cases for `withErrorBoundaryGroup` and `withAsyncBoundary`

## PR Checklist

- [x] I have written documents and tests, if needed.

### CODE REVIEW

File: packages/react/src/Delay.spec.tsx [hunk 1]
```diff
@@ -15,3 +15,15 @@ describe('Delay', () => {
...
+
+describe('withDelay', () => {
+  it('renders the children after the delay with component', async () => {
+    vi.useFakeTimers()
+    render(<TEXTAfterDelay100ms />)
+    expect(screen.queryByText(TEXT)).not.toBeInTheDocument()
+    act(() => vi.advanceTimersByTime(MS_100))
+    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument())
+  })
+})
```
Comment (okinawaa) [inline packages/react/src/Delay.spec.tsx:29]: cool 👍

File: packages/react/src/ErrorBoundaryGroup.spec.tsx [hunk 2]
```diff
@@ -76,3 +76,11 @@ describe('useErrorBoundaryGroup', () => {
...
   })
 })
+
+const TestWithErrorBoundaryGroup = withErrorBoundaryGroup(() => <>{TEXT}</>)
+describe('withErrorBoundaryGroup', () => {
+  it('should show children', () => {
+    const rendered = render(<TestWithErrorBoundaryGroup />)
+    expect(rendered.queryByText(TEXT)).toBeInTheDocument()
+  })
+})
```
Comment (okinawaa) [inline packages/react/src/ErrorBoundaryGroup.spec.tsx:86]: Can this represent the characteristics of ErrorBoundaryGroup?
I'm not sure if showing children is a big feature of ErrorBoundaryGroup!

File: packages/react/src/Suspense.spec.tsx [hunk 3]
```diff
@@ -57,3 +57,37 @@ describe('Suspense', () => {
     expect(screen.queryByText(TEXT)).toBeInTheDocument()
   })
 })
+
+const SuspendDuring100msToShowTEXTInSuspense = withSuspense(() => <Suspend during={MS_100} toShow={TEXT} />, {
+  fallback: FALLBACK,
+})
+const SuspendDuring100msToShowTEXTInSuspenseCSROnly = withSuspense.CSROnly(
```
Comment (okinawaa) [inline packages/react/src/Suspense.spec.tsx:64]: It's minor!

How about matching it with the actual component interface?
```suggestion
SuspendDuring100msToShowTEXTInSuspense.CSROnly = withSuspense.CSROnly(
```

**Review by okinawaa — APPROVED** [review]
Cool 😄

File: packages/react/src/ErrorBoundaryGroup.spec.tsx (hunk 2 above)
Comment (manudeli) [reply packages/react/src/ErrorBoundaryGroup.spec.tsx:86]: It's actually avoid codecov's failing, so I will improve in next Pull Request! Thanks!

File: packages/react/src/Suspense.spec.tsx (hunk 3 above)
Comment (manudeli) [reply packages/react/src/Suspense.spec.tsx:64]: Cool

Comment (manudeli) [reply packages/react/src/Suspense.spec.tsx:64]: When I apply this suggestion, I realized we need to type complexly. I'll try this after Pull Request

**Pull request merged**